android/java/org/chromium/chrome/browser/util/BraveProfileUtil.java

```diff
@@ -0,0 +1,33 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+package org.chromium.chrome.browser.util;
+
+import androidx.annotation.Nullable;
+
+import org.chromium.base.Log;
+import org.chromium.chrome.browser.app.BraveActivity;
+import org.chromium.chrome.browser.profiles.Profile;
+
+/** Util class that provides static methods to obtain Brave profiles. */
+public class BraveProfileUtil {
```

Comment:
this is problematic code that we should not be using at all so we don't want to extend it to any new code. Same thing applies to `activity.getCurrentProfile()` in general and anything that calls `ProfileManager.getLastUsedRegularProfile` or similar